pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -96,6 +96,16 @@ func ApplyResources(spec *v1beta1.TaskSpec, resolvedResources map[string]v1beta1
 	return ApplyReplacements(spec, replacements, map[string][]string{})
 }
 
+// ApplyContexts applies the substitution from $(context.(taskRun|task).*) with the specified values.
+// Currently supports only name substitution. Uses "" as a default if name is not specified.
+func ApplyContexts(spec *v1beta1.TaskSpec, rtr *ResolvedTaskResources, tr *v1beta1.TaskRun) *v1beta1.TaskSpec {
+	stringReplacements := map[string]string{}
+	stringReplacements["context.taskRun.name"] = tr.Name
+	stringReplacements["context.task.name"] = rtr.TaskName
```

Comment:
Just to double-check, if I'm defining a TaskRun's TaskSpec inline, there won't be a Task name, and this context value will be `""`, right?

Reply:
yes, and I added a testcase to confirm.